src/dotnet/commands/RestoringCommand.cs

```diff
@@ -20,7 +20,8 @@ private IEnumerable<string> ArgsToForwardToRestore()
                 !a.StartsWith("/t:") &&
                 !a.StartsWith("/target:") &&
                 !a.StartsWith("/ConsoleLoggerParameters:") &&
-                !a.StartsWith("/clp:"));
+                !a.StartsWith("/clp:") &&
+                !a.StartsWith("/p:TargetFramework"));
```

Comment:
I missed this in the original review, but we shouldn't be checking the args to forward this way. We should be forwarding all trailing args that are user-specified and any CLI-implicit args that are applicable to restore. This is going to cause inconsistent behavior between /p vs -p, /p vs. /property, etc.